apis/python/src/tiledbsoma/sparse_nd_array.py

```diff
@@ -179,8 +180,33 @@ def read_sparse_tensor(
         if format not in ("coo", "csr", "csc"):
             raise NotImplementedError("format not implemented")
 
+        # The full shape of the data is a template for the shape of the answer.
+        # For example if the data is 2D and coords is (None, [5,7,9]), then
+        # output shape[0] will be taken from the full shape, and output shape[1] will
+        # be 3.
         with self._tiledb_open("r") as A:
-            shape = A.shape
+            shape = list(A.shape)
+
+        for i in range(len(coords)):
```

Comment:
Rewriting the shape fairly significantly changes the behavior of this method, and I believe is not something we want to do.

Consider the following example:
* I have a sparse nd array that is shaped (100, 100)
* I call this method with:     `nda.read_sparse_tensor(( 8, 99 ), format="coo")`

This will return a SparseCOOTensor with:
* one value, at coordinate (8, 99)
* a shape of (1,1)
That doesn't make much sense.

I believe the original code was correct - the returned tensor should have the same shape as the soma.SparseNdArray, just fewer values.  This preserves the coordinate information, and keeps it self-consistent.

I'm also fairly certain that the current code will generate a SciPy error if you were to ask for `format="csr"`, as Scipy Sparse actually checks that coordinates are inside the shape limits.

Reply:
Also, the fact that this didn't fail in our unit tests highlights another case of where we need additional coverage!